torch/distributed/rpc/backend_registry.py

```diff
@@ -262,6 +262,8 @@ def _get_device_infos():
     agent = cast(TensorPipeAgent, api._get_current_rpc_agent())
     opts = agent._get_backend_options()
     device_count = torch.cuda.device_count()
+    if torch.cuda.is_available() and opts.devices:
+        torch.cuda.init()
```

Comment:
Is there a reason we add torch.cuda.init() here in addition to the call in the tensorpipe backend handler? Could we possibly have it in just one place?

Reply:
One is for statically sized groups, the other is for dynamic/elastic groups (devices are updated as workers join, therefore we need to do another cuda init if new devices are added) so they have different initialization paths and we can't put it in the same place for now.